dreadnode/cli/platform/download.py

```diff
@@ -1,144 +1,95 @@
     create_default_env_files,
 )
-from dreadnode.cli.platform.utils.printing import (
-    print_error,
+from dreadnode.cli.platform.tag import add_tag_arch_suffix
+from dreadnode.cli.platform.version import (
+    LocalVersion,
+    VersionConfig,
+)
+from dreadnode.logging_ import (
+    confirm,
     print_info,
     print_success,
-    print_warning,
-)
-from dreadnode.cli.platform.utils.versions import (
-    confirm_with_context,
-    create_local_latest_tag,
-    get_available_local_versions,
-    get_cli_version,
-    get_local_cache_dir,
 )
 
 
-def _resolve_latest(tag: str) -> str:
-    """Resolve 'latest' tag to actual version tag from API.
-
-    Args:
-        tag: Version tag that contains 'latest'.
-
-    Returns:
-        str: Resolved actual version tag.
+def download_platform(tag: str | None = None) -> LocalVersion:
     """
-    api_client = create_api_client()
-    release_info = api_client.get_platform_releases(
-        tag, services=[str(service) for service in SERVICES], cli_version=get_cli_version()
-    )
-    return release_info.tag
-
-
-def _create_local_version_file_structure(
-    tag: str, release_info: RegistryImageDetails
-) -> LocalVersionSchema:
-    """Create local file structure and update manifest for a new version.
+    Download platform version if not already available locally.
 
     Args:
-        tag: Version tag to create structure for.
-        release_info: Registry image details from API.
+        tag: Version tag to download (supports 'latest').
 
     Returns:
-        LocalVersionSchema: Created local version schema.
+        LocalVersionSchema: Local version schema for the downloaded/existing version.
     """
-    available_local_versions = get_available_local_versions()
-
-    # Create a new local version schema
-    local_cache_dir = get_local_cache_dir()
-    new_version = LocalVersionSchema(
-        **release_info.model_dump(),
-        local_path=local_cache_dir / tag,
-        current=False,
-    )
-
-    # Add the new version to the available local versions
-    available_local_versions.versions.append(new_version)
-
-    # sort the manifest by semver, newest first
-    available_local_versions.versions.sort(key=lambda v: v.tag, reverse=True)
-
-    # update the manifest file
-    manifest_path = local_cache_dir / VERSIONS_MANIFEST
-    with manifest_path.open(encoding="utf-8", mode="w") as f:
-        json.dump(available_local_versions.model_dump(), f, indent=2)
+    version_config = VersionConfig.read()
+    api_client = create_api_client()
 
-    print_success(f"Updated versions manifest at {manifest_path} with {new_version.tag}")
+    # 1 - Resolve the tag
 
-    if new_version.local_path.exists():
-        print_warning(f"Version {tag} already exists locally.")
-        if not confirm_with_context("overwrite it?"):
-            print_error("Aborting download.")
-            return new_version
+    tag = tag or "latest"
+    tag = add_tag_arch_suffix(tag)
```

Comment:
Tried to make tag selection a little more flexible. Users can pass a version with/without an arch.